src/vmm/src/vmm_config/boot_source.rs

```diff
@@ -102,7 +102,7 @@ impl BootConfig {
             Some(str) => str.as_str(),
         };
         cmdline
-            .insert_str(boot_args)
+            .prepend_str(boot_args)
```

Comment:
Nitish: We might get away with this because this step is peformed before adding the virtio devices. But in order to make absolutelty certain that everything is ok we would need to make sure that the  `boot_args` are actually appended and everything else is prepended. So do you think we should also have an `append_str()` method ?

Reply:
The cmdline is empty when the boot args are "prepended". This is actualy a workaround to the fact the Cmdline does not have a way of getting initialized to a certain value. 
However, I just gave it a second thought and it might be that the prepend method is not a very nice choice. Take a look at this: https://github.com/firecracker-microvm/firecracker/blob/200c2db7055e406125729c40e14acf47024c1420/src/vmm/src/builder.rs#L849.
We are appending the ro/rw to the string specifying the root device. With the new prepend method, we would need to first create the entire string root=dev/vda ro/rw and after that call into the Cmdline. IMO this makes things a little counterintuitive. 
So I am thinking of reverting to the previous insert_str version. WDYT?

Reply:
I see. There are other options. We can have 2 "regions" in the string:
- one for the firecracker custom args (virtio devices, etc) at the beginning
- one for the boot_args at the end
We can keep track of them by keeping the end index for each one. And have 2 groups of methods: 
- some that insert firecracker custom args. These would go in the first region.
- some that insert boot_args. These would go in the second region

Or there might be simpler methods to keep track of the regions. Or we can have 2 strings and concat them when `as_string` is called. It shouldn't add a significant overhead.

What do you think ?

Reply:
In rust-vmm we already started adding helper methods on top of `insert*` functions. For example, we have a function for [`add_virtio_mmio_device`](https://github.com/rust-vmm/linux-loader/blob/ff3ae8bd44a8f6b33713e4eeae1781751ff8c959/src/cmdline/mod.rs#L292).

Having a new function for `insert_boot_args` sounds like a good idea to me, as it is inline with existing helpers. This would come with the advantage that the existing `Cmdline` interface stays the same, so we do not introduce breaking changes for existing customers. With this, most of the changes will be in the implementation details of `Cmdline`.

In the future we would like to look at other improvements to `Cmdline` as well such as having it implemented with the builder pattern and allow even more custom/commonly used parameters as specified here: https://github.com/rust-vmm/linux-loader/issues/25.

Reply:
Nice info! ok, so since the direction in which rust-vmm's linux-loader has already shifted in the same way we want to go with this PR, I will open a PR there, merge it there and then bring the changes here. Thanks @andreeaflorescu for the heads up